shell/pages/diagnostic.vue

```diff
@@ -0,0 +1,439 @@
+<script>
+import { CAPI } from '@shell/config/types';
+import AsyncButton from '@shell/components/AsyncButton';
+import { downloadFile } from '@shell/utils/download';
+import { filterOnlyKubernetesClusters, filterHiddenLocalCluster } from '@shell/utils/cluster';
+import { sortBy } from '@shell/utils/sort';
+import { Checkbox } from '@components/Form/Checkbox';
+
+export default {
+  layout:     'plain',
+  components: { AsyncButton, Checkbox },
+  async fetch() {
+    const provClusters = await this.$store.dispatch('management/findAll', { type: CAPI.RANCHER_CLUSTER });
+    const readyClusters = provClusters.filter(c => c.mgmt?.isReady);
+    const clusterForCounts = filterHiddenLocalCluster(filterOnlyKubernetesClusters(readyClusters), this.$store);
+    const finalCounts = [];
+    const promises = [];
+    let topTenForResponseTime = [];
+
+    clusterForCounts.forEach((cluster, i) => {
+      finalCounts.push({
+        id:             cluster.id,
+        capiId:         cluster.mgmt?.id,
+        counts:         null,
+        isTableVisible: i === 0
+      });
+      promises.push(this.$store.dispatch('management/request', { url: `/k8s/clusters/${ cluster.mgmt?.id }/v1/counts` }));
+    });
+
+    const countsPerCluster = await Promise.all(promises);
+
+    countsPerCluster.forEach((clusterCount, index) => {
+      const counts = clusterCount.data?.[0]?.counts;
+
+      if (counts) {
+        const sanitizedCount = [];
+
+        Object.keys(counts).forEach((key) => {
+          sanitizedCount[key] = counts[key].summary?.count;
+          sanitizedCount.push({
+            resource: key,
+            count:    counts[key].summary?.count || 0,
+          });
+        });
+
+        const sortedCount = sortBy(sanitizedCount, 'count:desc');
+
+        topTenForResponseTime = topTenForResponseTime.concat(sortedCount);
+        topTenForResponseTime = sortBy(topTenForResponseTime, 'count:desc');
+
+        topTenForResponseTime.forEach((item, i) => {
+          topTenForResponseTime[i].id = finalCounts[index].id;
```

Comment:
Could we add these to sanitizedCount so we don't go over entries that already have them? (not that important if we're trimming topTenForResponseTime to 10 above)

Reply:
We shouldn't, otherwise we would be polluting the final count of resources in the JSON file (like we've seen on my demo). I removed those from the `sanitizedCount`. I have trimmed the top 10 response on every cluster iteration (new change from previous PR comment)